src/intree.js

```diff
@@ -128,15 +128,36 @@ module.exports.setup = function(cfg) {
           return false;
         }
 
+        let event = payload.details['event.type'];
         let events = task.task.extra.github.events;
-        let branches = task.task.extra.github.branches;
         let branch = payload.details['event.base.repo.branch'];
+        let includeBranches = task.task.extra.github.branches;
         let excludeBranches = task.task.extra.github.excludeBranches;
 
-        return _.some(events, ev => payload.details['event.type'].startsWith(_.trimEnd(ev, '*'))) &&
-          (!(branches || excludeBranches) || 
-            branches && _.includes(branches, branch) || excludeBranches && !_.includes(excludeBranches, branch) ||
-            payload.details['event.type']==='tag');
+        return _.some(events, ev => {
+          if (!event.startsWith(_.trimEnd(ev, '*'))) {
+            return false;
+          }
+
+          if (event !== 'push') {
+            return true;
+          }
+
+          if (includeBranches && excludeBranches) {
+            throw new Error(`Cannot include and exclude same branch. '${branch}' is in both lists!`);
+          }
```

Comment:
I think the error message here is incorrect.  The condition here is that there is both a `branches` and an `excludeBranches`.  The README suggests this situation is allowed (although I don't know what that would mean -- if a branch isn't on either list, is it included or excluded?).

I think when we talked about this briefly, we decided that this case is not allowed -- you must use no more than one of `branches` or `excludeBranches`.  Then there's no concern with the same branch appearing in both lists.